pkg/kubelet/secret/secret_manager.go

```diff
@@ -181,7 +184,7 @@ func (s *secretStore) Get(namespace, name string) (*v1.Secret, error) {
 		// Update state, unless we got error different than "not-found".
 		if err == nil || apierrors.IsNotFound(err) {
```

Comment:
I think this is slightly more readable then the logic you have. Feel free to disagree...
```go
if err != nil && !apierrors.IsNotFound(err) && (data.secret == nil && data.err == nil) {
    // Couldn't fetch the latest secret, but there is no cached data to return.
    // Return the fetch result instead.
    return secret, err
}

if (err == nil && !isSecretOlder(secret, data.secret)) || apierrorsIsNotFound(err){
    // If the fetch succeeded with a newer version of the secret, or if the secret could
    // not be found in the apiserver, update the cached data to reflect the current status.
    data.secret = secret
    data.err = err
    data.lastUpdateTime = s.clock.Now()
}

```